numpy/add_newdocs.py

```diff
@@ -1933,7 +1933,7 @@ def luf(lamdaexpr, *args, **kwargs):
     For 2-D arrays it is equivalent to matrix multiplication, and for 1-D
     arrays to inner product of vectors (without complex conjugation). For
     N dimensions it is a sum product over the last axis of `a` and
-    the second-to-last of `b`::
+    the second-to-last (or only, if `b` is 1-D) axis of `b`::
```

Comment:
Hmm, this still doesn't capture everything, and parenthesis should be avoided here. Maybe a list of possibilities: matrix-matrix, matrix-vector, vector-vector, scalar. Or maybe start with `ndim >= 2` -- stacked matrices, `ndim == 1` -- vector, `ndim == 0` -- scalar, and then make a list of combinations as below